### PR TITLE
[Tablet M2] Add Margins to Details View in Order Details

### LINKED ISSUE #10976: [Tablet M2] Add Margin to All Clickable Details in Order Details on Tablet
**Describe the bug**
In the app, specifically on a tablet, the order details section lacks sufficient margin around the shipping and billing details and others. This results in a cramped display of information, affecting the user interface and overall user experience.

**To Reproduce**
1. Open the WooCommerce app on a tablet.
2. Navigate to an Order and select 'Order Details'.
3. Click on 'Shipping Details' or 'Billing Details'.
4. Observe the lack of margin around the displayed information, in contrast to the order details section where margins are present.

https://github.com/woocommerce/woocommerce-android/assets/1509205/f0e38b5f-4bfd-4c83-9c22-c7c0b29a7ee2

**Expected behavior**
Shipping and billing details in the order details section should have appropriate margins similar to the order details section. This would ensure consistency across the UI and improve readability for users.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
- Device: Nexus 10
- WooCommerce Android version: `trunk`

### PR DESCRIPTION
**Pull request by jd-alexander**
Closes: #10976 

### Description

Leverages concepts utilized in https://github.com/woocommerce/woocommerce-android/pull/10949 to implement margins for all detail views within Order Details. 

- It implement details navigation controller logic.
- Let the details navigation controller keep its progress by not recreating it
- Order Details transition to a single pane for smaller screens. 

### Testing instructions

1. Open Orders on a tablet. 
2. Open an Order. 
3. Click on details such as Billing Address, Shipping Address, Customer Note and notice the margin. 
4. Go to the Products and ensure the details logic there isn't broken. 

### Images/gif

[order-details-margins_rotated.webm](https://github.com/woocommerce/woocommerce-android/assets/1509205/6d9a843d-a1be-47ad-945b-63a055477037)

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt [hunk 1]
```diff
@@ -111,6 +115,28 @@ class OrderListFragment :
...
+    override val twoPaneLayoutGuideline: Guideline
+        get() = binding.twoPaneLayoutGuideline
+    override val listPaneContainer: View
+        get() = binding.listPaneContainer
+    override val detailPaneContainer: View
+        get() = binding.detailPaneContainer
+    override val navigation: TabletLayoutSetupHelper.Screen.Navigation
+        get() = TabletLayoutSetupHelper.Screen.Navigation(
+            detailsNavGraphId = R.navigation.nav_graph_orders,
+            detailsInitialBundle = null
```
Comment (jd-alexander) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt:135]: @kidinov I want some initial thoughts from you here. Am I supposed to create an instance of `OrderDetailFragmentArgs` here and I am basically trying to figure out how to get the params for this. 

I see where `OrderListAdapter` handles the opening of the order detail hence the question as I explore this solution. 

Also what are your thoughts on the best verification for this functionality. I did the tests/steps you did in the product details.

**Comment by jd-alexander** [comment]
@kidinov in general let me know if I am on the right path in replicating the shared behavior for details on Products and Orders. 

I still have margin logic for the Order Details itself. Are we fine with that remaining?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt:135]: 👋

> I want some initial thoughts from you here. Am I supposed to create an instance of OrderDetailFragmentArgs here and I am basically trying to figure out how to get the params for this.

As you don't use `openItemDetails` method it's not used at all. If you wanna use that, then you need to pass the bundle (if any) you pas to the order details screen when it's open

Btw, not related to this. Why do you need `SelectedOrderTrackerViewModel`? I don't see that this VM shared between fragments at all. It seems to be selected order can be perfectly stored in `OrderListViewModel`

**Comment by kidinov** [comment]
@jd-alexander 

> in general let me know if I am on the right path in replicating the shared behavior for details on Products and Orders.

You are! Please notice that its `setDetailsMargins` is behind FF, which is supposed to be removed today (if all goes well)

> I still have margin logic for the Order Details itself. Are we fine with that remaining?

In my understanding, it should be removed, as the margin is set by `TabletLayoutSetupHelper` when all details fragments are created, so it looks like that it just overrides margins sets in the fragment

Look up `setDetailsMargins` method logic. In some case small adjustments to the layouts may be required to make it work

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt (hunk 1 above)
Comment (jd-alexander) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt:135]: > As you don't use openItemDetails method it's not used at all. If you wanna use that, then you need to pass the bundle (if any) you pas to the order details screen when it's open

Good point 

`SelectedOrderTrackerViewModel ` I came and saw this in the codebase 😅 It seems like an approach that can be optimized. Thanks for spotting 😁

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by jd-alexander** [comment]
@kidinov I think this should be ready now. 👍🏾 Tested it and the margins are being applied as expected.

**Comment by kidinov** [comment]
@jd-alexander Sorry, I didn't have time today to review. Will do tomorrow

**Review by kidinov — COMMENTED** [review]
The code changes looks good to me!

There is a bug, though; refund fragment has margins applied twice. That happens probably because that fragment itself embeds another fragment in it. Layout adjustments may help

image

Btw, IssueRefundFragment contains tons of unused since 2019 code 

```
// TODO: Temporary; it will be used again in a future release - do not remove
```

You could remove all of that if you'll touch that fragment

**Comment by jd-alexander** [comment]
> The code changes looks good to me!
> 
> There is a bug, though; refund fragment has margins applied twice. That happens probably because that fragment itself embeds another fragment in it. Layout adjustments may help
> 
> image
> Btw, IssueRefundFragment contains tons of unused since 2019 code
> 
> ```
> // TODO: Temporary; it will be used again in a future release - do not remove
> ```
> 
> You could remove all of that if you'll touch that fragment

Thanks for the review @kidinov this is the best way I thought of fixing the margins issue since the margin code wasn't contained within the individual fragments after investigating it https://github.com/woocommerce/woocommerce-android/pull/11011/commits/ba9e58847a0c0ae9a77780cae593fb9bf04121ea

For the code removal, I created a seperate PR to not make this one too big https://github.com/woocommerce/woocommerce-android/pull/11137

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/TabletLayoutSetupHelper.kt [hunk 2]
```diff
@@ -69,6 +71,13 @@ class TabletLayoutSetupHelper @Inject constructor(private val context: Context)
         )
     }
 
+    private fun shouldSetDetailsMargins(fragment: Fragment): Boolean {
+        return fragment != navHostFragment &&
+            fragment !is BottomSheetDialogFragment &&
+            fragment !is RefundByItemsFragment &&
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/TabletLayoutSetupHelper.kt:77]: Wdyt about using `fragment.parentFragment == navHostFragment` condition here?

**Review by kidinov — COMMENTED** [review]
@jd-alexander I notice that the margins are not applied to the custom fields fragment. Probably the same issue with layout, please take a look. Also, I left a suggesting regarding how we could apply margins only to the children not children of children =) Please take a look


https://github.com/woocommerce/woocommerce-android/assets/4923871/dd1690b7-676d-47d8-8044-6a644a8e8702

**Comment by jd-alexander** [comment]
> @jd-alexander I notice that the margins are not applied to the custom fields fragment. Probably the same issue with layout, please take a look. Also, I left a suggesting regarding how we could apply margins only to the children not children of children =) Please take a look
> 
>  03-22--11-52.mp4

Thanks! I will take a look.

**Comment by jd-alexander** [comment]
@samiuelson Let me know if you can take a look at this PR. The main new changes introduced are from this commit going down https://github.com/woocommerce/woocommerce-android/pull/11011/commits/35e5b8149c44c11782cac0c828dd0d128d11d636

The goal was to introduce logic that resolves this Compose related error https://github.com/woocommerce/woocommerce-android/pull/11011#pullrequestreview-1954507871

**Pull request closed without merging**